### ISSUE TITLE
Support for Ruby 2.7

### ISSUE THREAD

**Issue by marcotc** [grpc/grpc#21514]
### Is your feature request related to a problem? Please describe.
grpc gem fails to install under Ruby 2.7:

```
$ RBENV_VERSION=2.7.0-preview3 gem install grpc
Fetching grpc-1.25.0-universal-darwin.gem
Fetching google-protobuf-3.11.2-universal-darwin.gem
Fetching googleapis-common-protos-types-1.0.4.gem
Successfully installed google-protobuf-3.11.2-universal-darwin
Successfully installed googleapis-common-protos-types-1.0.4
ERROR:  Error installing grpc:
	The last version of grpc (>= 0) to support your Ruby & RubyGems was 1.25.0. Try installing it with `gem install grpc -v 1.25.0`
	grpc requires Ruby version >= 2.3, < 2.7.dev. The current ruby version is 2.7.0.preview3.
```

### Describe the solution you'd like
grpc gem should successfully install with Ruby 2.7.

### Describe alternatives you've considered
Sticking to Ruby < 2.7.

### Additional context
With the release of Ruby 2.7 expected to come later this month, support for it will soon become a need for the Ruby community.

**Comment by kapso** [comment]
Curious if support for `2.7.0` is coming anytime soon? Thanks

**Comment by brodock** [comment]
I've tried some smoke tests and it works if you run as platform ruby:

```
bundle lock --add-platform ruby
bundle config force_ruby_platform true
```

**Comment by nicolasnoble** [comment]
Duplicate of #21572.

**Comment by nicolasnoble** [comment]
Actually, this one's older, let's switch.

**Comment by nicolasnoble** [comment]
And as I mentioned in #21572, Christmas releases means there's nobody at the office to work on these for a few weeks.

**Comment by nicolasnoble** [comment]
Also blocked on rake-compiler/rake-compiler-dock#30

**Comment by davidstosik** [comment]
I don't want to urge anyone to release a new version of the grpc Ruby gem quickly, but while we're waiting on that release, how would one go and produce a gem that could be used locally to test an application, run automated tests, etc?

I've looked into [Build from source](https://github.com/grpc/grpc/tree/master/src/ruby#build-from-source) in the README but faced many problems, errors, for which I'm not prepared for. (The README also doesn't really tell me where I'd need to add 2.7 to the list of supported Ruby versions, I've tried grepping with little success...) Wonder if there's a guide, or a simplified procedure I could follow?

**Comment by Envek** [comment]
Christmas/New Year holidays is over in most locations, so I humbly hope that version of the grpc gem for Ruby 2.7 will be released soon. Thank you for supporting it :pray:

**Comment by Kjarrigan** [comment]
Any news? [rake-compiler/rake-compiler-dock v1.0.0](https://github.com/rake-compiler/rake-compiler-dock/releases/tag/1.0.0) was released on the 04.01. and supports 2.7 so this shouldn't be the blocking issue anymore, right?

**Comment by casperisfine** [comment]
Very hacky workaround I found:

At the top of your Gemfile:
```ruby
module BundlerHack
  def __materialize__
    if name == 'grpc' || name == 'google-protobuf'
      Bundler.settings.temporary(force_ruby_platform: true) do
        super
      end
    else
      super
    end
  end
end
Bundler::LazySpecification.prepend(BundlerHack)
```

This will force bundler to disregard the precompiled released of `grpc` & `google-protobuf`.

The source versions of the gems seem to compile just fine.

**Comment by dcoxall** [comment]
I have just tried to update grpc on the assumption that it now supports ruby 2.7

```
$ bundle update grpc
...
Gem::RuntimeRequirementNotMetError: grpc requires Ruby version >= 2.3, <
2.7.dev. The current ruby version is 2.7.0.0.
An error occurred while installing grpc (1.27.0), and Bundler
cannot continue.
Make sure that `gem install grpc -v '1.27.0' --source 'https://rubygems.org/'`
succeeds before bundling.

In Gemfile:
  grpc

$ ruby --version
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-darwin18]
```

**Comment by Envek** [comment]
@dcoxall, [as per rubygems.org](https://rubygems.org/gems/grpc) there were no new releases yet, so you have to either wait for release or use one of the hacks mentioned above.

**Comment by apolcyn** [comment]
Reopening to track until an official release has 2.7 support in binary packages. Sorry for the delay on this issue.

1.28.0.pre1 was just released, which has binary packages for linux and windows that can load on ruby 2.7 (mac binaries are missing from this release candidate, but will be added for the stable release).

**Comment by apolcyn** [comment]
`1.28.0.pre2` was just released, which also adds binaries for mac targeting ruby 2.7

**Comment by gnossen** [comment]
@apolcyn Has this been taken care of? Can this be closed out?

**Comment by apolcyn** [comment]
@gnossen yes I think so, outside of publishing 1.28.0 there shouldn't be more action needed on this.

**Comment by gnossen** [comment]
Going ahead and closing this out as a release blocker for 1.28.0 then.

**Comment by jjb** [comment]
@casperisfine You are a Scholar and a Gentleperson.

**Comment by jjb** [comment]
small modification to @casperisfine's recipe, so the hack only runs on macs (adding `if RUBY_PLATFORM =~ /darwin/`):

```ruby
if RUBY_PLATFORM =~ /darwin/
  module BundlerHack
    def __materialize__
      if name == 'grpc' || name == 'google-protobuf'
        Bundler.settings.temporary(force_ruby_platform: true) do
          super
        end
      else
        super
      end
    end
  end
  Bundler::LazySpecification.prepend(BundlerHack)
end
```

**Comment by michelson** [comment]
in lates bundler you can do this: 
`gem "grpc", force_ruby_platform: true`
